src/tribler/core/utilities/tests/test_pony_utils.py

```diff
@@ -120,3 +124,109 @@ def test_format_warning():
+    with pytest.raises(ZeroDivisionError):
+        with marking_corrupted_db(db_path):
+            raise ZeroDivisionError()
+
+    assert not Path(str(db_path) + '.is_corrupted').exists()
+
+
+def test_marking_corrupted_db__malformed(db_path: Path):
+    with pytest.raises(DatabaseIsCorrupted):
+        with marking_corrupted_db(db_path):
+            raise sqlite3.DatabaseError('database disk image is malformed')
+
+    assert Path(str(db_path) + '.is_corrupted').exists()
+
+
+def test_get_db_version__db_does_not_exist(tmp_path: Path):
+    db_path = tmp_path / 'doesnotexist.db'
+
+    with pytest.raises(RuntimeError, match='^The version value is not found in database .*doesnotexist.db$'):
+        get_db_version(db_path)
+
+
+def test_get_db_version__db_does_not_exist_default_version(tmp_path: Path):
+    db_path = tmp_path / 'doesnotexist.db'
+
+    default_version = 123
+    version = get_db_version(db_path, default=default_version)
+    assert version == default_version
+
+
+def test_get_db_version__version_table_does_not_exist(db_path: Path):
+    with pytest.raises(RuntimeError, match='^The version value is not found in database .*test.db$'):
+        get_db_version(db_path)
+
+
+def test_get_db_version__version_table_does_not_exist_default_version(db_path: Path):
+    default_version = 123
+    version = get_db_version(db_path, default=default_version)
+    assert version == default_version
+
+
+def test_get_db_version(db_path: Path):
+    with sqlite3.connect(db_path) as connection:
+        connection.execute('create table MiscData(name text primary key, value text)')
+        connection.execute("insert into MiscData(name, value) values('db_version', '100')")
+
+    version = get_db_version(db_path)
+    assert version == 100
+
+    version = get_db_version(db_path, default=99)
+    assert version == 100
+
+    version = get_db_version(db_path, default=101)
+    assert version == 100
+
+
+def test_get_db_version__corrupted_db(tmp_path: Path):
+    db_path = tmp_path / 'test.db'
+
+    connection = sqlite3.connect(db_path)
+    connection.execute('create table MiscData(name text primary key, value text)')
+    connection.execute("insert into MiscData(name, value) values('db_version', '100')")
+    connection.commit()
+    assert table_exists(connection.cursor(), 'MiscData')
+    connection.close()
+
+    marker_path = Path(str(db_path) + '.is_corrupted')
+    marker_path.touch()
+
+    default_version = 10
+    version = get_db_version(db_path, default=default_version)
+    assert version == default_version
+
+    with sqlite3.connect(db_path) as connection:
+        assert not table_exists(connection.cursor(), 'MiscData')
+
+
+def test_mark_db_as_corrupted_file_does_not_exist(tmp_path: Path):
+    db_path = tmp_path / 'doesnotexist.db'
+    with pytest.raises(RuntimeError, match='^Corrupted database file not found: .*doesnotexist.db$'):
+        _mark_db_as_corrupted(db_path)
```

Comment:
Could you please add descriptions to the tests?

Reply:
I added descriptions to all tests